python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -330,6 +339,47 @@ def select_operator_to_run(
     )
 
 
+def _try_to_scale_up_cluster(topology: Topology):
+    """Try to scale up the cluster to accomodate the provided in-progress workload.
+
+    This makes a resource request to Ray's autoscaler consisting of the current,
+    aggregate usage of all operators in the DAG + the incremental usage of all operators
+    that are ready for dispatch (i.e. that have inputs queued). If the autoscaler were
+    to grant this resource request, it would allow us to dispatch one task for every
+    ready operator.
+
+    Note that this resource request does not take the global resource limits or the
+    liveness policy into account; it only tries to make the existing resource usage +
+    one more task per ready operator feasible in the cluster.
+
+    Args:
+        topology: The execution state of the in-progress workload for which we wish to
+            request more resources.
+    """
+    # Get resource usage for all ready ops.
+    usage = ExecutionResources()
+    for op, state in topology.items():
+        usage = usage.add(op.current_resource_usage())
+        # Only include incremental resource usage for ops that are ready for
+        # dispatch.
+        if state.num_queued() > 0:
+            # TODO(Clark): Scale up more aggressively by adding incremental resource
+            # usage for more than one bundle in the queue for this op?
+            usage = usage.add(op.incremental_resource_usage())
+    resource_request = {}
+    # Round usage to ints, since autoscaler only accepts integral resource requests.
+    if usage.cpu:
+        resource_request["CPU"] = round(usage.cpu)
+    if usage.gpu:
+        resource_request["GPU"] = round(usage.gpu)
+    if usage.object_store_memory:
```

Comment:
Shall we avoid requesting object store memory? I don't think this is well supported / commonly used for autoscaling. It might just completely not work if node configs don't have object store memory as a provided resource. Also, ops don't report this in incremental.

Reply:
Dropped.